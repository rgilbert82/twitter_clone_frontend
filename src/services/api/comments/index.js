export { default as getCommentsAPI } from './getComments.js';
export { default as createCommentAPI } from './createComment.js';
export { default as editCommentAPI } from './editComment.js';
export { default as deleteCommentAPI } from './deleteComment.js';
