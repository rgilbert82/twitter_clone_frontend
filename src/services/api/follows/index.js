export { default as getFollowAPI } from './getFollow.js';
export { default as createFollowAPI } from './createFollow.js';
export { default as deleteFollowAPI } from './deleteFollow.js';
export { default as getUserFollowersAPI } from './getUserFollowers.js';
export { default as getUserFollowingsAPI } from './getUserFollowings.js';
