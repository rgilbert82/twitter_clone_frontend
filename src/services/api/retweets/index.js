export { default as createRetweetAPI } from './createRetweet.js';
export { default as deleteRetweetAPI } from './deleteRetweet.js';
