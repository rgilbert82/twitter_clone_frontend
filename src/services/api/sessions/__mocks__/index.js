export { default as loginAPI } from './login.js';
export { default as logoutAPI } from './logout.js';
export { default as getCurrentUserAPI } from './getCurrentUser.js';
