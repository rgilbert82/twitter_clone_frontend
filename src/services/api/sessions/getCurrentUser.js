import { get_API_URL } from '../../misc';

export default () => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/sessions/1`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(JSON.parse(request.responseText));
        } else {
          reject(request.responseText);
        }
      }
    };

    request.open('GET', path);
    request.withCredentials = true;
    request.setRequestHeader('Authorization', document.cookie || '');
    request.send();
  });
}
