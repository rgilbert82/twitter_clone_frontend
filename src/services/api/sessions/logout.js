import { get_API_URL } from '../../misc';

export default () => {
  return new Promise((resolve, reject) => {
    const path = `${get_API_URL()}/sessions/1`;
    let request = new XMLHttpRequest();

    request.onreadystatechange = () => {
      if (request.readyState === XMLHttpRequest.DONE) {
        if (request.status === 200) {
          resolve(request.responseText);
        } else {
          reject(request.responseText);
        }
      }
    };

    request.open('DELETE', path);
    request.withCredentials = true;
    request.setRequestHeader('Authorization', document.cookie || '');
    request.send();
  });
}
