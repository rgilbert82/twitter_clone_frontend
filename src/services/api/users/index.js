export { default as getUsersAPI } from './getUsers.js';
export { default as getUserAPI } from './getUser.js';
export { default as createUserAPI } from './createUser.js';
export { default as updateUserAPI } from './updateUser.js';
export { default as getUserTweetsAPI } from './getUserTweets.js';
export { default as deleteUserAPI } from './deleteUser.js';
