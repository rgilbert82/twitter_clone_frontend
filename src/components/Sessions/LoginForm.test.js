jest.mock('../../services/api/sessions');

import React from 'react';
import { LoginForm } from '.';
import { shallow } from 'enzyme';
import { loginAPI } from '../../services/api/sessions';

describe(LoginForm, () => {
  let appLoginMock = jest.fn();
  let component;

  beforeEach(() => {
    component = shallow(<LoginForm appLogin={appLoginMock} />);
  });

  it('renders properly', () => {
    expect(component).toMatchSnapshot();
  });

  it('updates username on input', () => {
    expect(component.state('username')).toEqual('');
    component.find('#username_field').simulate('change', { target: { value: 'User123' } });
    expect(component.state('username')).toEqual('User123');
  });

  it('updates password on input', () => {
    expect(component.state('password')).toEqual('');
    component.find('#password_field').simulate('change', { target: { value: 'password' } });
    expect(component.state('password')).toEqual('password');
  });

  it('logs in the user', () => {
    expect.assertions(3);
    component.setState({ username: 'RG', password: 'password' });
    return component.instance().submitForm()
      .then(() => {
        expect(component.state('username')).toEqual('');
        expect(component.state('password')).toEqual('');
        expect(appLoginMock).toBeCalled();
      });
  });
});
