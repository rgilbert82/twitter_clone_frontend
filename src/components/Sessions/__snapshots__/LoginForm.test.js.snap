// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class LoginForm extends _react2.default.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '' };


    this.updateUsername = this.updateUsername.bind(this);
    this.updatePassword = this.updatePassword.bind(this);
    this.submitForm = this.submitForm.bind(this);
  }

  updateUsername(e) {
    this.setState({ username: e.target.value });
  }

  updatePassword(e) {
    this.setState({ password: e.target.value });
  }

  submitForm() {
    const loginData = this.state;
    this.setState({ username: '', password: '' });

    return (0, _sessions.loginAPI)(loginData).
    then(data => {
      this.props.appLogin(data.token);
    }).catch(() => {
      this.props.displayMessage('Incorrect username / password');
    });
  }

  render() {
    let component =
    _react2.default.createElement(_reactBootstrap.Navbar.Form, { id: 'login_form', pullRight: true, __source: { fileName: _jsxFileName, lineNumber: 40 }, __self: this },
      _react2.default.createElement(_reactBootstrap.FormGroup, { __source: { fileName: _jsxFileName, lineNumber: 41 }, __self: this },
        _react2.default.createElement(_reactBootstrap.FormControl, { id: 'username_field', type: 'text', onChange: this.updateUsername, value: this.state.username, placeholder: 'Username', __source: { fileName: _jsxFileName, lineNumber: 42 }, __self: this })),

      _react2.default.createElement(_reactBootstrap.FormGroup, { __source: { fileName: _jsxFileName, lineNumber: 44 }, __self: this },
        _react2.default.createElement(_reactBootstrap.FormControl, { id: 'password_field', type: 'password', onChange: this.updatePassword, value: this.state.password, placeholder: 'Password', __source: { fileName: _jsxFileName, lineNumber: 45 }, __self: this })),

      _react2.default.createElement(_reactBootstrap.Button, { onClick: this.submitForm, disabled: !this.state.username || !this.state.password, __source: { fileName: _jsxFileName, lineNumber: 47 }, __self: this }, 'Sign In'));



    return component;
  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm
    appLogin={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <FormGroup
          bsClass="form-group"
>
          <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    id="username_field"
                    onChange={[Function]}
                    placeholder="Username"
                    type="text"
                    value=""
          />
</FormGroup>,
        <FormGroup
          bsClass="form-group"
>
          <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    id="password_field"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
</FormGroup>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={true}
          onClick={[Function]}
>
          Sign In
</Button>,
      ],
      "componentClass": "div",
      "id": "login_form",
      "pullLeft": false,
      "pullRight": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form-group",
          "children": <FormControl
            bsClass="form-control"
            componentClass="input"
            id="username_field"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-control",
            "componentClass": "input",
            "id": "username_field",
            "onChange": [Function],
            "placeholder": "Username",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form-group",
          "children": <FormControl
            bsClass="form-control"
            componentClass="input"
            id="password_field"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-control",
            "componentClass": "input",
            "id": "password_field",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Sign In",
          "disabled": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Sign In",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <FormGroup
            bsClass="form-group"
>
            <FormControl
                        bsClass="form-control"
                        componentClass="input"
                        id="username_field"
                        onChange={[Function]}
                        placeholder="Username"
                        type="text"
                        value=""
            />
</FormGroup>,
          <FormGroup
            bsClass="form-group"
>
            <FormControl
                        bsClass="form-control"
                        componentClass="input"
                        id="password_field"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
</FormGroup>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={true}
            onClick={[Function]}
>
            Sign In
</Button>,
        ],
        "componentClass": "div",
        "id": "login_form",
        "pullLeft": false,
        "pullRight": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": <FormControl
              bsClass="form-control"
              componentClass="input"
              id="username_field"
              onChange={[Function]}
              placeholder="Username"
              type="text"
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-control",
              "componentClass": "input",
              "id": "username_field",
              "onChange": [Function],
              "placeholder": "Username",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": <FormControl
              bsClass="form-control"
              componentClass="input"
              id="password_field"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-control",
              "componentClass": "input",
              "id": "password_field",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Sign In",
            "disabled": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Sign In",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
